### ISSUE TITLE
Snowflake SEMANTIC_VIEW queries parse as generic functions and trigger RF01

### ISSUE THREAD

**Issue by jaysobel** [sqlfluff/sqlfluff#8446]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The Snowflake dialect does not have first-class grammar for the [`SEMANTIC_VIEW(...)`](https://docs.snowflake.com/en/sql-reference/constructs/semantic_view) query construct. A simple invocation is accepted through generic function grammar, so the semantic-view object name and logical-table-qualified fields are represented as ordinary references. RF01 then reports false positives. Adding the construct's internal `WHERE` clause produces an unparsable section.

I searched issues, pull requests, source, changelog, and git history and found no existing query-grammar implementation or report for `SEMANTIC_VIEW`.

### Expected Behaviour

Snowflake `SEMANTIC_VIEW(...)` relations should parse into intentional segments for the semantic-view object, metrics, facts, dimensions, aliases, and predicate. RF01 should ignore semantic identifiers inside the construct while continuing to check genuine relational references in the surrounding query.

### Observed Behaviour

Given:

```sql
SELECT
    order_date,
    revenue
FROM SEMANTIC_VIEW(
    analytics.semantic.sales_analysis
    DIMENSIONS orders.order_date
    METRICS orders.revenue
);
```

SQLFluff 4.3.0 parses `SEMANTIC_VIEW(...)` as a generic `function`. RF01 reports three false positives: the semantic-view object plus both semantic fields. Adding the documented internal predicate `WHERE orders.order_date >= DATE '2025-01-01'` causes a PRS violation for the parenthesized semantic-view body.

### How to reproduce

Save the SQL example above as `semantic_view.sql`, then run:

```console
sqlfluff parse semantic_view.sql --dialect snowflake
sqlfluff lint semantic_view.sql --dialect snowflake --rules RF01
```

To reproduce the PRS behavior, add this line after the `METRICS` clause:

```sql
WHERE orders.order_date >= DATE '2025-01-01'
```

### Dialect

Snowflake

### Version

SQLFluff 4.3.0; Python 3.10.13. The behavior is also present on current `main` at `dd429de04190d336090e5880f1b54dfd6730039d`.

### Configuration

No project configuration. The commands specify `--dialect snowflake` and, for reference analysis, `--rules RF01`.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
